restapi/src/main/java/io/stargate/web/docsapi/service/ReactiveDocumentService.java

```diff
@@ -489,12 +498,238 @@ public ReactiveDocumentService(
   }
 
   // we need to transform the stuff to support array elements
-  private List<String> processSubDocumentPath(List<String> subDocumentPath) {
+  public List<String> processSubDocumentPath(List<String> subDocumentPath) {
     return subDocumentPath.stream()
         .map(path -> DocsApiUtils.convertArrayPath(path))
         .collect(Collectors.toList());
   }
 
+  /**
+   * Retrieves the specified path in the document as an array, appends the @param value, and returns
+   * what the array will look like after the append. Note that this doesn't write the array back to
+   * the data store.
+   *
+   * @param db
+   * @param namespace
+   * @param collection
+   * @param documentId
+   * @param processedPath
+   * @param value
+   * @param context
+   * @return a JSON representation of the array after pushing the new value
+   */
+  public Maybe<JsonNode> getArrayAfterPush(
+      DocumentDB db,
+      String namespace,
+      String collection,
+      String documentId,
+      List<String> processedPath,
+      Object value,
+      ExecutionContext context) {
+    return getDocument(db, namespace, collection, documentId, processedPath, null, context)
+        .map(
+            array -> {
+              JsonNode data = array.getData();
+              if (!data.isArray()) {
+                throw new ErrorCodeRuntimeException(
+                    ErrorCode.DOCS_API_SEARCH_ARRAY_PATH_INVALID,
+                    "The path provided to push to has no array");
+              }
+              ArrayNode arrayData = (ArrayNode) data;
+              if (value == null) {
```

Comment:
isn't there anything available to overcome this crazy if/else sequence? @tatu-at-datastax suggestions?

Reply:
I'll look into this, there's definitely just a method that I don't know about on the mapper or something

Reply:
Unfortunately I don't think there is anything super-convenient, and would probably prefer explicit traversal of Map/List over use of `valueToTree()` just to be fully explicit (plus there's a bit of overhead with conversion as logically that serializes value as tokens, reads back -- nothing too bad but maybe not worth it here) so there is nothing funny going on (in particular, ensure they are all "simple" values, Numbers, Strings, Booleans, Maps, Lists, no POJOs).

But I would extract this into a helper method, with recursive calls for `Map`, `List`.

Reply:
`arrayNode.insertPOJO(...)` seems to have worked here... is that the wrong way to do this though?

Reply:
Ah I see what you wrote about `no POJOs` - I'll revert this

Reply:
Interesting idea! `insertPOJO` will add "opaque" value which will be serialized just fine for normal usage, as if serialized on its own. So yes, it should work; if there were issues a simple test would catch it. That methods just adds "POJONode" wrapper which should be fine. It'd be awkward if `JsonNode` was processed etc but when constructing payload seems like a reasonable choice.
Good thinking!

Reply:
And just realized my "no POJOs" simply meant that if converting from Lists/Maps, it's simplest if there were no other types: but `insertPOJO` / `addPOJO` (for ArrayNode) are perfectly legit for those too. And `Map`s and `List`s etc; anything Jackson can serialize as JSON value is fine when `JsonNode` structure is just for that purpose.